source/adapters/level_zero/command_buffer.cpp

```diff
@@ -933,7 +933,8 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
       MustSignalWaitEvent = false;
     }
   }
-  if (MustSignalWaitEvent) {
+  if (MustSignalWaitEvent &&
+      !CommandBuffer->WaitEvent->CounterBasedEventsEnabled) {
```

Comment:
If we didn't set the `CounterBasedEventEnabled` parameter to true when `WaitEvent` is [created](https://github.com/oneapi-src/unified-runtime/blob/4ef258c85ea39c5c5b4269ee9ff1768a0d11192b/source/adapters/level_zero/command_buffer.cpp#L438)
```cpp
UR_CALL(EventCreate(Context, nullptr, false, false,
                      &RetCommandBuffer->WaitEvent));
```
Why do we need to check here that `CounterBasedEventsEnabled` isn't set? Can this property of the event change after the event has been created?